lib/rails_pg_adapter/patch.rb

```diff
@@ -13,34 +13,55 @@ module Patch
       "the database system is starting up",
       "connection is closed",
       "could not connect",
+      "is not currently accepting connections",
     ].freeze
     CONNECTION_ERROR_RE = /#{CONNECTION_ERROR.map { |w| Regexp.escape(w) }.join("|")}/.freeze
 
     CONNECTION_SCHEMA_ERROR = ["PG::UndefinedColumn"].freeze
     CONNECTION_SCHEMA_RE = /#{CONNECTION_SCHEMA_ERROR.map { |w| Regexp.escape(w) }.join("|")}/.freeze
 
-    CONNECTION_READ_ONLY_ERROR = [
-      "read-only",
-      "PG::ReadOnlySqlTransaction",
```

Comment:
Do we handle RDS failovers if we remove this? I thought this was of the common errors that fired during that

Reply:
yep, we do. from here: https://github.com/tines/rails-pg-adapter/blob/0eb6f03b7aa8c4ca85114e239895fb42ed5b5229/lib/rails_pg_adapter/patch.rb#L10

This just introduced to only check if a read-only error was raised (if say, we blocked writes on the DB for a cut over) and then do a dedicate reconnect for those. It was introduced here: https://github.com/tines/rails-pg-adapter/commit/b1250b315f5c9be89ae206115b1491ac5fbf1d45

That approach didn't quite work out for extended periods and left the connection in a messy state. So, instead during failover, connection disruptions or anything similar, we just disconnect and remove the connection, and then retry. Or, re-raise the error so it gets bubbled up.